web/js/students_of_class.js

```diff
@@ -245,172 +302,224 @@ function constructStudentTableRow(student) {
-        return false;
-    }
+        });
+    await fetchApi();
 }
 
 function editOrAdd() {
-    const editOption = document.getElementById("EditOrAdd");
+    const editOption = document.querySelector("#EditOrAdd");
     if (editOption.value === "edit") {
         return true;
     } else if (editOption.value === "add") {
         return false;
     }
 }
 
-function editStudent() {
-    modalEditStudent.modal('hide');
+async function editStudent(formData) {
     const errorElement = document.querySelector("#error").value;
-    const editFirstName = document.getElementById("editFirstName").value;
-    const editLastName = document.getElementById("editLastName").value;
-    const editBirthday = document.getElementById("editBirthday").value;
-    const editFemale = stillFemale();
-    const edits = {firstName: editFirstName, lastName: editLastName, birthDay: editBirthday, female: editFemale};
-    const studentURL = document.getElementById("studentURL").value;
-    patchStudent(studentURL, edits)
+    modalEditStudent.modal('hide');
+    const data = {
+        firstName: formData.get("editFirstName"),
+        lastName: formData.get("editLastName"),
+        birthDay: formData.get("editBirthDay"),
+        female: formData.get("editFemale") === "female"
+    };
+    const studentURL = formData.get("studentURL");
+    await patchStudent(studentURL, data)
         .catch(() => {
             errorElement.innerHTML = "The patch request was not successful.";
             $(errorElement).slideDown().delay(3000).slideUp();
-        })
-}
-
-function ShowHideSaveButton(promise) {
-    promise ? document.getElementById("saveSportResultButton").style.visibility = 'visible' : document.getElementById("saveSportResultButton").style.visibility = 'hidden';
+        });
+    await fetchApi();
 }
 
 function updateSchoolClass(schoolClass) {
-    const className = document.getElementById("class-name");
+    const className = document.querySelector("#class-name");
     className.innerHTML = `${schoolClass.grade}${schoolClass.className}`;
 
-    const classTitle = document.getElementById("title");
+    const classTitle = document.querySelector("#title");
     classTitle.innerHTML = `${schoolClass.grade}${schoolClass.className}`;
 
-    const classBreadCrumb = document.getElementById("breadCrumb");
+    const classBreadCrumb = document.querySelector("#breadCrumb");
     classBreadCrumb.innerHTML = `${schoolClass.grade}${schoolClass.className}`;
 
-    const classTeacher = document.getElementById("class-teacher");
+    const classTeacher = document.querySelector("#class-teacher");
     classTeacher.innerHTML = schoolClass.classTeacherName ?? '';
 
-    document.getElementById("classURL").value = schoolClass._links.self.href;
+    document.querySelector("#classURL").value = schoolClass._links.self.href;
+}
+
+function addSportResults() {
+    const table = document.getElementsByClassName("value");
+    const students = document.getElementsByClassName("student");
```

Comment:
If you would use `document.querySelectorAll(".student")` you would save the `Array.from`.